src/main/java/games/strategy/triplea/ai/AIUtils.java

```diff
@@ -80,7 +80,7 @@ private static ProductionRule getProductionRule(final UnitType unitType, final P
    *        - calculate the strength of the units in a sea or land battle?
    */
   public static float strength(final Collection<Unit> units, final boolean attacking, final boolean sea) {
-    int strength = 0;
+    float strength = 0;
```

Comment:
This really seems to be programmer mistale error prone correctly pointed out

Reply:
Agreed, this looks like an obvious oversight.  However, it's been there since Sean added it in f2fb4ea9 (circa 2006), so the Weak AI has been operating with this "algorithm" since then.  The question is will this introduce an unexpected change in the behavior of the Weak AI?

Note also that while most callers treat the return value of this method as a `float`, there are a few that store it in an `int`.  We may want to address those concurrently.  In which case, a separate PR just to correct this may be justified so it's easy to track in the history instead of buried with other unrelated changes.

@ron-murhammer Do you foresee any drastic changes in the Weak AI with this fix?  The only lines that would seem to be affected are line 92:

```java
        // two hit
        strength += 1.5 * unitAttachment.getHitPoints();
```

and line 103:

```java
          // a unit with attack of 0 isnt worth much
          // we dont want transports to try and gang up on subs
          if (unitAttachment.getAttack(u.getOwner()) == 0) {
            strength -= 1.2;
          }
```

In the first case, it looks like the strength could be off in the range [0,0.5], while in the second case, in the range [-0.2,0].  Let's just say that's ±1.  That doesn't seem like much, but I don't have a good feeling for what a normal strength is in the Weak AI.

Reply:
Should be fine. Honestly, weak AI is so random that I don't think anyone would be able to notice. Some day I want to remove all the weak AI code anyways but would have to have a much faster version of fast AI in order to do it.